pkg/defkinds/nodejs/definition.go

```diff
@@ -0,0 +1,283 @@
+package nodejs
+
+import (
+	"github.com/NiR-/zbuild/pkg/builddef"
+	"github.com/NiR-/zbuild/pkg/defkinds/webserver"
+	"github.com/NiR-/zbuild/pkg/llbutils"
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+)
+
+var defaultBaseImages = map[string]string{
+	"10": "docker.io/library/node:10-buster-slim",
+	"12": "docker.io/library/node:12-buster-slim",
+	"13": "docker.io/library/node:13-buster-slim",
+}
+var supportedVersions = "10, 12, 13"
+
+func defaultDefinition() Definition {
+	isDev := true
+	isNotDev := false
+	healthcheckEnabled := true
+
+	return Definition{
+		BaseStage: Stage{
+			ExternalFiles:  []llbutils.ExternalFile{},
+			SystemPackages: map[string]string{},
+			GlobalPackages: map[string]string{},
+			ConfigFiles:    map[string]string{},
+			Healthcheck:    &healthcheckEnabled,
+		},
+		BaseImage: "",
+		Stages: map[string]DerivedStage{
+			"dev": {
+				DeriveFrom: "base",
+				Dev:        &isDev,
+			},
+			"prod": {
+				DeriveFrom: "base",
+				Dev:        &isNotDev,
```

Comment:
it's always strange to use negative vars :thinking: 

let's say I'm not in dev context, so the `is not dev` affirmation is verified, so I'll tend to write `isNotDev = true`. But then, `true` will be passed to `dev: &isNotDev`, whereas it's supposed to get the opposite :thinking: 

What about renaming these vars (also for the other kind definitions) to `isDev / isProd`, or `devStageDevMode / prodStageDevMode` ?

Or maybe only use a single `isDev` var ?

Reply:
> it's always strange to use negative vars

:100: It confuses me too :|

> What about renaming these vars (also for the other kind definitions) to isDev / isProd, or devStageDevMode / prodStageDevMode ?
>
> Or maybe only use a single isDev var ?

I can't use a single var since these struct props are scalar pointers but your first proposal looks great, I'll try that :wink:

Reply:
I'll fix var names in a separate PR as this one is big enough.